pkg/tar/tar.go

```diff
@@ -38,10 +38,10 @@ var ErrNotSupportedPlatform = errors.New("platform and architecture is not suppo
 // root of the tar file and should be cleaned (for example using filepath.Clean)
 type PathWhitelistMap map[string]struct{}
 
-// extractTar extracts a tarball (from a tar.Reader) into the root directory
-// if pwl is not nil, only the paths in the map are extracted.
-// If overwrite is true, existing files will be overwritten.
-func extractTar(tr *tar.Reader, overwrite bool, pwl PathWhitelistMap, uidRange *uid.UidRange) error {
+// ExtractTarInsecure extracts a tarball (from a tar.Reader) into the root
+// directory if pwl is not nil, only the paths in the map are extracted. If
+// overwrite is true, existing files will be overwritten.
```

Comment:
I think running this as non-root might tease out some more edge cases I haven't thought of but we can roll forward with that.